Add Falcon-7B deepspeed example

es94129 opened the pull request:
Run on 4 A10 GPUs: go/dogfood#notebook/4210361116937762

lu-wang-dl commented on llm-examples/falcon/scripts/fine_tune_deepspeed_falcon.py:
```diff
@@ -0,0 +1,242 @@
+ROOT_PATH = Path(__file__).parent.parent
+MODEL_PATH = "tiiuae/falcon-7b"
+TOKENIZER_PATH = "tiiuae/falcon-7b"
+DEFAULT_TRAINING_DATASET = "timdettmers/openassistant-guanaco"
```
Why do we want to use this dataset instead of our dolly one?

es94129 replied:
The blog example of Falcon used `timdettmers/openassistant-guanaco` for fine-tuning (https://huggingface.co/blog/falcon) so it might be something users would try.

Also the question-answer pairs in databricks-dolly-15k do not have much difference from the dataset that Falcon-7B was trained on, which leads to the case that the original Falcon-7B could already answer them pretty well, so fine-tuning doesn't seem useful in that case. The openassistant dataset had different languages from Falcon data at least, which is easier to see the effect of fine-tuning.

lu-wang-dl replied:
`tiiuae/falcon-7b` is a base model which are trained on a diverse range of tasks and data, unlike `tiiuae/falcon-7b-instruct` which is an instruction model that optimized for specific tasks or domains. `tiiuae/falcon-7b-instruct` is finetuned on a mixture of chat/instruct datasets.

I think our task is to fine tune the base model `tiiuae/falcon-7b` with databricks-dolly-15k to make it like the `tiiuae/falcon-7b-instruct`.

lu-wang-dl replied:
https://oliviermills.com/articles/understanding-ai-models-base-language-learning-models-vs-instruction-tuned-language-learning-models

es94129 commented:
Rebased so the commit history wasn't clear; changes since yesterday:
- Use the `a10_config.json` file from https://github.com/databricks/databricks-ml-examples/pull/12
- Batch size increased to 16, `gradient_accumulation_steps=8`
- Evaluation strategy changed to `epoch`
- Updated the GPU requirements in README and the notebook cell
